core/src/main/java/org/ensime/core/javac/TreeUtilities.java

```diff
@@ -0,0 +1,1077 @@
+            }
+
+            @Override
+            public Void visitMethod(MethodTree node, Void p) {
+                int[] span = findNameSpan(node);
+                
+                if (span != null && span[0] <= pos && pos < span[1]) {
+                    throw new Result(getCurrentPath());
+                }
+                
+                return super.visitMethod(node, p);
+            }
+        }
+        
+        try {
+            new PathFinder(pos, sourcePositions).scan(path, null);
+        } catch (Result result) {
+            path = result.path;
+        }
+        
+        if (path.getLeaf() == path.getCompilationUnit())
+            return path;
+        
+        TokenSequence<JavaTokenId> tokenList = tokensFor(path.getLeaf(), sourcePositions);
+        tokenList.moveEnd();
+        if (tokenList.movePrevious() && tokenList.offset() < pos) {
+            switch (tokenList.token().id()) {
+                case GTGTGT:
+                case GTGT:
+                case GT:
+                    if (path.getLeaf().getKind() == Tree.Kind.MEMBER_SELECT || TreeUtilities.CLASS_TREE_KINDS.contains(path.getLeaf().getKind()) || path.getLeaf().getKind() == Tree.Kind.GREATER_THAN)
+                        break;
+                case RPAREN:
+                    if (path.getLeaf().getKind() == Tree.Kind.ENHANCED_FOR_LOOP || path.getLeaf().getKind() == Tree.Kind.FOR_LOOP ||
+                            path.getLeaf().getKind() == Tree.Kind.IF || path.getLeaf().getKind() == Tree.Kind.WHILE_LOOP ||
+                            path.getLeaf().getKind() == Tree.Kind.DO_WHILE_LOOP || path.getLeaf().getKind() == Tree.Kind.TYPE_CAST)
+                        break;
+                case SEMICOLON:
+                    if (path.getLeaf().getKind() == Tree.Kind.FOR_LOOP &&
+                            tokenList.offset() <= sourcePositions.getStartPosition(path.getCompilationUnit(), ((ForLoopTree)path.getLeaf()).getUpdate().get(0)))
+                        break;
+                case RBRACE:
+                    path = path.getParentPath();
+                    switch (path.getLeaf().getKind()) {
+                        case CATCH:
+                            path = path.getParentPath();
+                        case METHOD:
+                        case FOR_LOOP:
+                        case ENHANCED_FOR_LOOP:
+                        case SYNCHRONIZED:
+                        case WHILE_LOOP:
+                        case TRY:
+                            path = path.getParentPath();
+                            break;
+                        case IF:
+                            do {
+                                path = path.getParentPath();
+                            } while (path != null && path.getLeaf().getKind() == Tree.Kind.IF);
+                            break;
+                    }
+                    break;
+            }
+        }
+        return path;
+    }
+    
+//    /**Parses given type in given context.
```

Comment:
Did you comment this out, or was it already like this?  If you took it out I would add a comment at the top with where the code came from (after the 'DO NOT REMOVE' notice, and then just delete the commented code.